examples/pipeline_wavernn/wavernn_inference_wrapper.py

```diff
@@ -0,0 +1,259 @@
+        r'''Applies a crossfade and unfolds into a 1d array.
+
+        y = [[seq1],
+             [seq2],
+             [seq3]]
+        Apply a gain envelope at both ends of the sequences
+        y = [[seq1_in, seq1_timesteps, seq1_out],
+             [seq2_in, seq2_timesteps, seq2_out],
+             [seq3_in, seq3_timesteps, seq3_out]]
+        Stagger and add up the groups of samples:
+            [seq1_in, seq1_timesteps, (seq1_out + seq2_in), seq2_timesteps, ...]
+
+        Args:
+            y (Tensor): Batched sequences of audio samples with shape
+                (num_folds, timesteps + 2 * overlap).
+            overlap (int): Timesteps for both xfade and rnn warmup.
+
+        Returns:
+            unfolded waveform (Tensor) : waveform in a 1d tensor with shape (total_len).
+        '''
+
+        num_folds, length = y.shape
+        timesteps = length - 2 * overlap
+        total_len = num_folds * (timesteps + overlap) + overlap
+
+        # Need some silence for the rnn warmup
+        silence_len = overlap // 2
+        fade_len = overlap - silence_len
+        silence = torch.zeros((silence_len), dtype=y.dtype, device=y.device)
+        linear = torch.ones((silence_len), dtype=y.dtype, device=y.device)
+
+        # Equal power crossfade
+        t = torch.linspace(-1, 1, fade_len, dtype=y.dtype, device=y.device)
+        fade_in = torch.sqrt(0.5 * (1 + t))
+        fade_out = torch.sqrt(0.5 * (1 - t))
+
+        # Concat the silence to the fades
+        fade_in = torch.cat([silence, fade_in])
+        fade_out = torch.cat([linear, fade_out])
+
+        # Apply the gain to the overlap samples
+        y[:, :overlap] *= fade_in
+        y[:, -overlap:] *= fade_out
+
+        unfolded = torch.zeros((total_len), dtype=y.dtype, device=y.device)
+
+        # Loop to add up all the samples
+        for i in range(num_folds):
+            start = i * (timesteps + overlap)
+            end = start + timesteps + 2 * overlap
+            unfolded[start:end] += y[i]
+
+        return unfolded
+
+    def _pad_tensor(self, x: Tensor, pad: int, side: str = 'both') -> Tensor:
+        r"""Pad the given tensor.
+
+        Args:
+            x (Tensor): The tensor to pad with shape (n_batch, n_mels, time).
+            pad (int): The amount of padding applied to the input.
+
+        Return:
+            padded (Tensor): The padded tensor with shape (n_batch, n_mels, time).
+        """
+        b, c, t = x.size()
+        total = t + 2 * pad if side == 'both' else t + pad
+        padded = torch.zeros((b, c, total), device=x.device)
+        if side == 'before' or side == 'both':
+            padded[:, :, pad:pad + t] = x
+        elif side == 'after':
+            padded[:, :, :t] = x
+        else:
+            raise ValueError(f"Unexpected side: '{side}'. "
+                            f"Valid choices are 'both', 'before' and 'after'.")
+        return padded
+
+    @torch.jit.export
+    def infer(self,
```

Comment:
I think you can call this method `forward` in the wrapper, and no need to decorate `torch.jit.export`.

Reply:
Addressed in #1650